packages/design-system/src/scalars/components/fragments/textarea-field/textarea-field.test.tsx

```diff
@@ -0,0 +1,232 @@
+      expect(textarea).toHaveClass(
+        "border-red-700",
+        "bg-red-50/50",
+        "dark:border-red-400",
+        "dark:bg-red-900/5",
+      );
+
+      // Custom class
+      rerender(
+        <TextareaField
+          // eslint-disable-next-line tailwindcss/no-custom-classname
+          className="custom-class"
+        />,
+      );
+      expect(textarea).toHaveClass("custom-class");
+    });
+  });
+
+  describe("Messages and Feedback", () => {
+    it("should handle warnings and errors", () => {
+      render(
+        <TextareaField
+          description="Help text"
+          warnings={["Warning 1", "Warning 2"]}
+          errors={["Error 1", "Error 2"]}
+        />,
+      );
+
+      expect(screen.getByText("Help text")).toBeInTheDocument();
+      expect(screen.getByText("Warning 1")).toBeInTheDocument();
+      expect(screen.getByText("Warning 2")).toBeInTheDocument();
+      expect(screen.getByText("Error 1")).toBeInTheDocument();
+      expect(screen.getByText("Error 2")).toBeInTheDocument();
+    });
+  });
+
+  /* describe("Input Handling", () => {
+    it("should handle various input methods and transformations", async () => {
+      const user = userEvent.setup();
+      const onChange = vi.fn();
+
+      render(
+        <TextareaField
+          value="Initial"
+          onChange={onChange}
+          maxLength={20}
+          trim
+          uppercase
+        />,
+      );
+
+      const textarea = screen.getByRole("textbox");
+
+      // Regular typing
+      await user.type(textarea, " text");
+      // Check the actual textarea value
+      expect(textarea).toHaveValue("INITIAL TEXT");
+
+      // Emoji input
+      await user.clear(textarea);
+      await user.type(textarea, "👋🏻");
+      expect(textarea).toHaveValue("👋🏻");
+      expect(screen.getByText("2/10")).toBeInTheDocument();
+
+      // Paste handling
+      const pasteData = "Pasted";
+      const pasteEvent = new ClipboardEvent("paste", {
+        clipboardData: new DataTransfer(),
+      });
+      pasteEvent.clipboardData?.setData("text/plain", pasteData);
+      textarea.dispatchEvent(pasteEvent);
+
+      // RTL text handling
+      await user.clear(textarea);
+      const rtlText = "مرحبا";
+      await user.type(textarea, rtlText);
+      expect(textarea).toHaveValue(rtlText);
+      expect(textarea).toHaveStyle({ direction: "rtl" });
+    });
+  }); */
+
+  /* describe("Accessibility", () => {
+    it("should handle all accessibility attributes and announcements", () => {
+      render(
+        <TextareaField
+          errors={["Error message"]}
+          maxLength={10}
+          value="Test"
+        />,
+      );
+
+      const textarea = screen.getByRole("textbox");
+      const counter = screen.getByText("4/10");
+
+      expect(textarea).toHaveAttribute("aria-label", "Text area");
+      expect(textarea).toHaveAttribute("aria-invalid", "true");
+      expect(counter).toBeInTheDocument();
+    });
+  }); */
```

Comment:
Please, remove this code, you can access it through the git history

Reply:
Done